samples/matter/common/src/bt_nus_service.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_nus_service.h"
+
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#include <zephyr/settings/settings.h>
+
+using namespace ::chip;
+using namespace ::chip::DeviceLayer;
+
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
+
+namespace
+{
+#if (CONFIG_BT_FIXED_PASSKEY)
+constexpr uint32_t kDefaultPasskey = 123456;
+#endif
+constexpr uint32_t kAdvertisingOptions = BT_LE_ADV_OPT_CONNECTABLE;
+constexpr uint8_t kAdvertisingFlags = BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR;
+} // namespace
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = NUSService::Connected,
+	.disconnected = NUSService::Disconnected,
+	.security_changed = NUSService::SecurityChanged,
+};
+struct bt_conn_auth_cb NUSService::sConnAuthCallbacks = {
+	.passkey_display = AuthPasskeyDisplay,
+	.cancel = AuthCancel,
+};
+struct bt_conn_auth_info_cb NUSService::sConnAuthInfoCallbacks = {
+	.pairing_complete = PairingComplete,
+	.pairing_failed = PairingFailed,
+};
+struct bt_nus_cb NUSService::sNusCallbacks = {
+	.received = RxCallback,
+};
+
+NUSService NUSService::sInstance;
+
+CHIP_ERROR NUSService::Init(const char *name, size_t nameLen, uint8_t priority, uint16_t minInterval,
+			    uint16_t maxInterval)
+{
+	VerifyOrReturnError(name && nameLen > 0, CHIP_ERROR_INVALID_ARGUMENT);
+
+	mAdvertisingItems[0] = BT_DATA(BT_DATA_FLAGS, &kAdvertisingFlags, sizeof(kAdvertisingFlags));
+	mAdvertisingItems[1] = BT_DATA(BT_DATA_NAME_COMPLETE, name, static_cast<uint8_t>(nameLen));
+
+	mServiceItems[0] = BT_DATA(BT_DATA_UUID128_ALL, BT_UUID_NUS_VAL, sizeof(BT_UUID_NUS_VAL));
+
+	mAdvertisingRequest.priority = priority;
+	mAdvertisingRequest.options = kAdvertisingOptions;
+	mAdvertisingRequest.minInterval = minInterval;
+	mAdvertisingRequest.maxInterval = maxInterval;
+	mAdvertisingRequest.advertisingData = Span<bt_data>(mAdvertisingItems);
+	mAdvertisingRequest.scanResponseData = Span<bt_data>(mServiceItems);
+
+	mAdvertisingRequest.onStarted = [](int rc) {
+		if (rc == 0) {
+			ChipLogDetail(DeviceLayer, "NUS BLE advertising started");
```

Comment:
Ideally we should not use ChipLog here, but Zephyr's LOG_... macro. But yeah, I know in other files from common we use ChipLog as well...

Reply:
But still we should stick to the Zephyr logger in new code I guess :)